lib/stdlib/src/uri_string.erl

```diff
@@ -375,6 +375,80 @@ recompose(Map) ->
     end.
 
 
+%%-------------------------------------------------------------------------
+%% Resolve URIs
+%%-------------------------------------------------------------------------
+-spec resolve(RefURI, BaseURI) -> TargetURI when
+      RefURI :: uri_string() | uri_map(),
+      BaseURI :: uri_string() | uri_map(),
+      TargetURI :: uri_string()
+                 | error().
+resolve(URIMap, BaseURIMap) ->
+    resolve(URIMap, BaseURIMap, []).
+
+
+-spec resolve(RefURI, BaseURI, Options) -> TargetURI when
+      RefURI :: uri_string() | uri_map(),
+      BaseURI :: uri_string() | uri_map(),
+      Options :: [return_map],
+      TargetURI :: uri_string() | uri_map()
+                 | error().
+resolve(URIMap, BaseURIMap, Options) ->
+    try resolve1(URIMap, BaseURIMap, Options)
+    catch
+        throw:{error, Atom, RestData} -> {error, Atom, RestData}
+    end.
+
+resolve1(URI, BaseURI, Options) ->
+	TargetURI = case resolve1(ensure_map_uri(URI), BaseURI) of
+		TargetURI0 = #{path := Path0} ->
+			%% We remove dot segments. Normalizing the entire URI
```

Comment:
The extra slash is added by the scheme based normalization see: 

```
6.2.3.  Scheme-Based Normalization

For example,
   because the "http" scheme makes use of an authority component, has a
   default port of "80", and defines an empty path to be equivalent to
   "/", the following four URIs are equivalent:

      http://example.com
      http://example.com/
      http://example.com:/
      http://example.com:80/

   In general, a URI that uses the generic syntax for authority with an
   empty path should be normalized to a path of "/".
```
I think it is safe to normalize the whole URI here!
See my comment below regarding the failing TC.

Reply:
I am no longer using the (full) normalize function, it's now using the `normalize_path_segment` function which just calls `remove_dot_segments` on the path. The comment is no longer relevant as a result.